### DISCUSSION TITLE
nextcloud-aio update fails

### DISCUSSION THREAD

**Discussion by nowatbuer** [nextcloud/all-in-one#4306] [Questions] [answered]
I am running Nextcloud AIO v7.12.1 on ubuntu 23.10. since a few days. Now Nextcloud informed about a new mastercontainer available for update. I started the update but it failed with the following logs:

time="2024-03-02T10:28:22Z" level=debug msg="Sleeping for a second to ensure the docker api client has been properly initialized."
time="2024-03-02T10:28:23Z" level=debug msg="Making sure everything is sane before starting"
time="2024-03-02T10:28:23Z" level=info msg="Watchtower 1.7.1"
time="2024-03-02T10:28:23Z" level=info msg="Using no notifications"
time="2024-03-02T10:28:23Z" level=info msg="Only checking containers which name matches \"nextcloud-aio-mastercontainer\""
time="2024-03-02T10:28:23Z" level=info msg="Running a one time update."
time="2024-03-02T10:28:23Z" level=debug msg="Checking containers for updated images"
time="2024-03-02T10:28:23Z" level=debug msg="Retrieving running containers"
time="2024-03-02T10:28:23Z" level=debug msg="Trying to load authentication credentials." container=/nextcloud-aio-mastercontainer image="nextcloud/all-in-one:latest"
time="2024-03-02T10:28:23Z" level=debug msg="No credentials for index.docker.io found" config_file=/config.json
time="2024-03-02T10:28:23Z" level=debug msg="Got image name: nextcloud/all-in-one:latest"
time="2024-03-02T10:28:23Z" level=debug msg="Checking if pull is needed" container=/nextcloud-aio-mastercontainer image="nextcloud/all-in-one:latest"
time="2024-03-02T10:28:23Z" level=debug msg="Built challenge URL" URL="https://index.docker.io/v2/"
time="2024-03-02T10:28:23Z" level=debug msg="Got response to challenge request" header="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\"" status="401 Unauthorized"
time="2024-03-02T10:28:23Z" level=debug msg="Checking challenge header content" realm="https://auth.docker.io/token" service=registry.docker.io
time="2024-03-02T10:28:23Z" level=debug msg="Setting scope for auth token" image=docker.io/nextcloud/all-in-one scope="repository:nextcloud/all-in-one:pull"
time="2024-03-02T10:28:23Z" level=debug msg="No credentials found."
time="2024-03-02T10:28:24Z" level=debug msg="Parsing image ref" host=index.docker.io image=nextcloud/all-in-one normalized=docker.io/nextcloud/all-in-one tag=latest
time="2024-03-02T10:28:24Z" level=debug msg="Doing a HEAD request to fetch a digest" url="https://index.docker.io/v2/nextcloud/all-in-one/manifests/latest"
time="2024-03-02T10:28:24Z" level=debug msg="Found a remote digest to compare with" remote="sha256:96c38da308ce9e2c863743cba2678f504b52c5834c6eba6ffbbae939b45c028c"
time="2024-03-02T10:28:24Z" level=debug msg=Comparing local="sha256:0648d91c1689f00c75e6fd38c5ce51f71b3ae34d8e47e34951060b60624024e7" remote="sha256:96c38da308ce9e2c863743cba2678f504b52c5834c6eba6ffbbae939b45c028c"
time="2024-03-02T10:28:24Z" level=debug msg="Digests did not match, doing a pull."
time="2024-03-02T10:28:24Z" level=debug msg="Pulling image" container=/nextcloud-aio-mastercontainer image="nextcloud/all-in-one:latest"
time="2024-03-02T10:28:25Z" level=info msg="Found new nextcloud/all-in-one:latest image (7813294e71d8)"
time="2024-03-02T10:28:25Z" level=info msg="Stopping /nextcloud-aio-mastercontainer (4ca95967f5bf) with SIGTERM"
time="2024-03-02T10:28:25Z" level=error msg="Error response from daemon: Cannot kill container: 4ca95967f5bf5bf3ed277ea5dcb9cccfd6c9edd1bbca9975ee337d43e0144fbd: Cannot kill container 4ca95967f5bf5bf3ed277ea5dcb9cccfd6c9edd1bbca9975ee337d43e0144fbd: unknown error after kill: runc did not terminate successfully: exit status 1: unable to signal init: permission denied\n: unknown"
time="2024-03-02T10:28:25Z" level=info msg="Session done" Failed=1 Scanned=1 Updated=0 notify=no
time="2024-03-02T10:28:25Z" level=info msg="Waiting for the notification goroutine to finish" notify=no

a manual 
docker stop 4ca95967f5bf5bf3ed277ea5dcb9cccfd6c9edd1bbca9975ee337d43e0144fbd 
succeeds, but does not help me further. 4ca95... is the nextcloud-aio-mastercontainer

**Comment by szaimen** [comment]
Hi, can you post the output of `sudo docker info` here?

**Comment by nowatbuer** [comment]
Don't know, whether it is relevant or not: I have deployed that installation via ansible-playbook.
Here the output from docker info:
sudo docker info
Client:
 Version:    24.0.5
 Context:    default
 Debug Mode: false

Server:
 Containers: 22
  Running: 21
  Paused: 0
  Stopped: 1
 Images: 57
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 
 runc version: 
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.5.0-21-generic
 Operating System: Ubuntu 23.10
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.27GiB
 Name: cloud
 ID: 43a12701-7725-480a-81b9-98e540feff7e
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by szaimen** [comment]
Hm... Can you upgrade docker via `sudo apt update && sudo apt upgrade` and try the update again?

**Comment by nowatbuer** [comment]
Everything is up to date:
sudo apt update && sudo apt upgrade
OK:1 http://de.archive.ubuntu.com/ubuntu mantic InRelease
OK:2 http://de.archive.ubuntu.com/ubuntu mantic-updates InRelease
OK:3 http://de.archive.ubuntu.com/ubuntu mantic-backports InRelease                  
OK:4 http://security.ubuntu.com/ubuntu mantic-security InRelease                     
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
Alle Pakete sind aktuell.
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
Paketaktualisierung (Upgrade) wird berechnet… Fertig
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.

Btw, I managed to get around that update problem by stopping the mastercontainer manually and remove it via docker rm. Finally reinstalling nextcloud-aio via ansible-playbook started the updated version. 

In my original posting there is a line:
time="2024-03-02T10:28:25Z" level=error msg="Error response from daemon: Cannot kill container:
For me it seems to have to do with missing permissions. Which account must be enabled to stop/kill containers?

**Comment by szaimen** [comment, marked as answer]
> "Error response from daemon: Cannot kill container: 4ca95967f5bf5bf3ed277ea5dcb9cccfd6c9edd1bbca9975ee337d43e0144fbd: Cannot kill container 4ca95967f5bf5bf3ed277ea5dcb9cccfd6c9edd1bbca9975ee337d43e0144fbd: unknown error after kill: runc did not terminate successfully: exit status 1: unable to signal init: permission denied\n: unknown"

This looks like a bug in watchtower: https://github.com/containrrr/watchtower/issues/1891

**Reply by nowatbuer** [reply]
Thanks, I will wait for a fix.
